lom/Makefile

```diff
@@ -61,97 +64,164 @@ format_code:
+echo -e "Detail file coverage from $$1 :"
+echo "_____________________________________________________________________________________"
+file_coverage="$$(sed -n 's/.*<option value="[^"]\+">\([^<]\+\).* (\([^)]\+\%.*\)<\/option>.*/\1 (\2)/p' "$$1")"; 
+echo "$${file_coverage}"
+echo "====================================================================================="
 endef
+export extract_file_coverage
+
+add-extract_file_coverage: 
+	$(shell mkdir -p $(BUILD_DIR))
+	$(shell rm -f  $(BUILD_DIR)/extract_file_coverage.sh)
+	$(shell echo >> $(BUILD_DIR)/extract_file_coverage.sh)
+	@echo "$$extract_file_coverage" >> $(BUILD_DIR)/extract_file_coverage.sh
+	$(shell chmod 777 $(BUILD_DIR)/extract_file_coverage.sh 2>/dev/null || :)
+
+# Clean up the script file
+clean-extract_file_coverage:
+	rm -f $(BUILD_DIR)/extract_file_coverage.sh
+
+# Define the aggregate file coverage extraction function
+define aggregate_file_coverage
+#!/bin/bash
+echo "====================================================================================="
+echo -e "Aggregate file coverages : "
+echo "_____________________________________________________________________________________"
+for file in $(TEST_DIR)/*.html; do
+    if [ -f "$$file" ]; then
+        file_coverage=$$(sed -n 's/.*<option value="[^"]\+">\([^<]\+\).* (\([^)]\+\%.*\)<\/option>.*/\1 (\2)/p' "$$file")
+        echo "$$file_coverage"
+    fi
+done
+echo "_____________________________________________________________________________________"
+endef
+export aggregate_file_coverage
+
+# Print all the source fiules coverages at one place
+add-aggregate_file_coverage: 
+	$(shell mkdir -p $(BUILD_DIR))
+	$(shell rm -f  $(BUILD_DIR)/aggregate_file_coverage.sh)
+	$(shell echo >> $(BUILD_DIR)/aggregate_file_coverage.sh)
+	@echo "$$aggregate_file_coverage" >> $(BUILD_DIR)/aggregate_file_coverage.sh
+	$(shell chmod 777 $(BUILD_DIR)/aggregate_file_coverage.sh 2>/dev/null || :)
+
+# Clean up the script file
+clean-aggregate-file_coverage:
+	rm -f $(BUILD_DIR)/aggregate_file_coverage.sh
+
+.PHONY: all clean format_code clean-extract_file_coverage add-extract_file_coverage clean-aggregate-file_coverage add-aggregate_file_coverage COMMON_DEPS
+
+COMMON_DEPS := format_code add-extract_file_coverage add-aggregate_file_coverage $(CMN_C_LIB) $(ENGINE_BIN) $(PLUGIN_MANAGER_BIN) $(LIB_TEST) $(ENGINE_TEST) \
+	$(PLUGIN_MANAGER_TEST) $(SONIC_DBCLIENT_TEST) $(SONIC_LINK_CRC_TEST) $(PLUGINS_COMMON_TEST) \
+	$(PY_TEST)
 
-.PHONY: all clean format_code
+all: $(COMMON_DEPS)
+	$(BUILD_DIR)/aggregate_file_coverage.sh
 
-all: format_code $(CMN_C_LIB) $(ENGINE_BIN) $(LIB_TEST) $(ENGINE_TEST) $(PLUGIN_MANAGER_TEST) \
-	$(SONIC_DBCLIENT_TEST) $(SONIC_LINK_CRC_TEST) $(PLUGINS_COMMON_TEST) $(PY_TEST) \
-	$(PLUGIN_MANAGER_BIN)
+all-silent: LIB_TEST_FLAGS :=
+all-silent: $(COMMON_DEPS)
+	$(BUILD_DIR)/aggregate_file_coverage.sh
 
+all-skiptests: format_code $(CMN_C_LIB) $(ENGINE_BIN) $(PLUGIN_MANAGER_BIN)
+	$(BUILD_DIR)/aggregate_file_coverage.sh
```

Comment:
I don't see a need for this target. During dev time, we can always create custom targets for our needs. For example, I would never like to skip Lib tests.

Reply:
cleaned up